slides-03-design.qmd

```diff
@@ -0,0 +1,364 @@
+
+```{r}
+#| eval: false
+as_survey_rep(
+  .data,
+  variables = NULL, # defaults to all in .data
+  repweights = NULL, # Variables specifying the replication weights
+  weights = NULL, # Variable specifying the analytic/main weight
+  type = c(
+    "BRR", "Fay", "JK1", "JKn", "bootstrap", "successive-difference", "ACS",
+    "other"
+  ), # Type of replication weight
+  combined_weights = TRUE, # TRUE if repweights already include sampling weights, usually TRUE
+  rho = NULL, # Shrinkage factor for Fay's method
+  bootstrap_average = NULL, # For type="bootstrap", if the bootstrap weights have been averaged, gives the number of iterations averaged over
+  scale = NULL, # Scaling constant for variance
+  rscales = NULL, # Scaling constants for variance
+  mse = getOption("survey.replicates.mse"), # If TRUE, compute variance based around point estimates rather than mean of replicates
+  degf = NULL, # Design degrees of freedom, otherwise calculated based on number of replicate weights
+)
+```
+
+## Syntax for common replicate methods {.smaller}
+
+```{r}
+#| eval: FALSE
+#| code-line-numbers: "1-5|7-12|14-19|21-26"
+
+brr_des <- dat %>%
+  as_survey_rep(
+    weights = WT,
+    repweights = starts_with("REPWT"),
+    type = "BRR",
+    mse = TRUE
+  )
+
+fay_des <- dat %>%
+  as_survey_rep(
+    weights = WT0,
+    repweights = num_range("WT", 1:20),
+    type = "Fay",
+    mse = TRUE,
+    rho = 0.3
+  )
+
+jkn_des <- dat %>%
+  as_survey_rep(
+    weights = WT0,
+    repweights = WT1:WT20,
+    type = "JKN",
+    mse = TRUE,
+    rscales = rep(0.1, 20)
+  )
+
+bs_des <- dat %>%
+  as_survey_rep(
+    weights = pw,
+    repweights = pw1:pw50,
+    type = "bootstrap",
+    scale = 0.02186589,
+    mse = TRUE
+  )
+```
+
+- Note: this uses fake data and can't be run, just syntax examples
+
+## Example: RECS 2020 {.smaller}
+
+- [Using the microdata file to compute estimates and relative standard errors](https://www.eia.gov/consumption/residential/data/2020/pdf/microdata-guide.pdf)
+
+> The following instructions are examples for calculating any RECS estimate using the final weights
+(NWEIGHT) and the associated RSE using the replicate weights (NWEIGHT1 – NWEIGHT60).
+
+- Includes R syntax for {survey} package which gets us what we need for {srvyr}
+
+```{r}
+#| eval: false
+repweights <- select(RECS2020, NWEIGHT1:NWEIGHT60)
+RECS <- svrepdesign(
+  data = RECS2020,
+  weight = ~NWEIGHT,
+  repweights = repweights,
+  type = "JK1",
+  combined.weights = TRUE,
+  scale = (ncol(repweights) - 1) / ncol(repweights),
+  mse = TRUE
+)
+```
+
+## Example: RECS 2020 Syntax {.smaller}
+
+```{r}
+#| label: design-recs-example
+
+recs_des <- recs_2020 %>%
+  as_survey_rep(
+    weight = NWEIGHT, #<1>
+    repweights = NWEIGHT1:NWEIGHT60, #<2>
+    type = "JK1", #<3>
+    scale = 59 / 60, #<4>
+    mse = TRUE
+  ) #<5>
+
+summary(recs_des) #<6>
```

Comment:
Move this to the next slide as it's off the bottom of the page and focus on the code here.